synapse_auto_accept_invite/__init__.py

```diff
@@ -28,6 +32,22 @@ def __init__(self, config: InviteAutoAccepterConfig, api: ModuleApi):
         self._api = api
         self._config = config
 
+        should_run_on_this_worker = (
+            config.worker_to_run_on is None and self._api.worker_name is None
+        ) or (config.worker_to_run_on == self._api.worker_name)
```

Comment:
I think this is just checking equality?

```suggestion
        should_run_on_this_worker = (
            config.worker_to_run_on == self._api.worker_name
        )
```

Reply:
I seem to remember it being bad practice to use `==` with `None` but when talking about strings I can't see the problem, so sure

Reply:
> Definitely do not use `is` with strings:

yes indeed, this is the other way around though.
I don't know why, maybe just dogma, but I seem to have been told that `== None` is Wrong and `is None` is Right.
I don't see why as `x is None` ⇒  `x == None` (but not the other way around, but for strings I doubt a string would claim itself equal to `None`).
/shrug :-)

Reply:
Oh I see. Should have read the context.

I think it's largely dogma/convention/style. https://stackoverflow.com/questions/3257919/what-is-the-difference-between-is-none-and-none points has some context, but nothing particularly terrifying. (If you're in a situation where `== None` has different behaviour to `is None`, you have bigger problems).

Reply:
Usually `is None` is what you want to use, but in this case we're comparing if two variables are equal, not to a constant.